include/cliffordsynthesis/CliffordSynthesizer.hpp

```diff
@@ -0,0 +1,289 @@
+/*
+* This file is part of the MQT QMAP library which is released under the MIT license.
+* See file README.md or go to https://iic.jku.at/eda/research/ibm_qx_mapping/ for more information.
+*/
+
+#ifndef QMAP_CLIFFORDSYNTHESIS_H
+#define QMAP_CLIFFORDSYNTHESIS_H
+
+#include "Architecture.hpp"
+#include "CliffordOptimizationResult.hpp"
+#include "Encodings/Encodings.hpp"
+#include "LogicBlock/LogicBlock.hpp"
+#include "QuantumComputation.hpp"
+#include "operations/OpType.hpp"
+#include "operations/StandardOperation.hpp"
+
+#include <algorithm>
+#include <bitset>
+#include <cassert>
+#include <chrono>
+#include <cmath>
+#include <functional>
+#include <iostream>
+#include <istream>
+#include <iterator>
+#include <list>
+#include <memory>
+#include <mutex>
+#include <set>
+#include <sstream>
+#include <thread>
+#include <tuple>
+#include <unordered_set>
+#include <utility>
+#include <vector>
+
+class CliffordOptimizer {
+public:
+    CliffordOptimizer() = default;
+    void init(bool pchoose_best = false, bool puse_embedding = false, unsigned char pnqubits = 0, unsigned short pinitial_timesteps = 0, SynthesisStrategy strategy = SynthesisStrategy::UseMinimizer, SynthesisTarget ptarget = SynthesisTarget::GATES);
+    void setCircuit(const qc::QuantumComputation& qc);
+    void setTableau(Tableau& targetTabl);
```

Comment:
In addition to these methods it would actually make sense to provide constructors for the optimizer.
One taking a a circuit + all the configuration options
And one taking a target tableau + all configurations.
Both of these might optionally take an initial tableau as well.
In a similar spirit, they might optionally take an architecture.